misk/web/tabs/loader/src/sagas/itemSaga.ts

```diff
@@ -0,0 +1,86 @@
+// In case you need to use a selector
+// import also select from redux-saga/effects
+// and then simplie yield select(yourSelector())
+//
+// In case you need to redirect to whatever route
+// import { push } from react-router-redux and then
+// yield put(push('/next-page'))
+
+import axios from "axios"
+import { all, call, put, takeLatest } from "redux-saga/effects"
+
+import {
+  IAction, IActionType, ITEM, dispatchItem
+} from "../actions"
+
+function * handleGet () {
```

Comment:
Are all these supposed to be generators in the typical sense? It kind of looks like it's using generators in place of async/await.

Reply:
tldr; yes, but generators provide additional features useful in writing tests. We're using the redux-saga flow initially to try and provide a bit of structure for requests, actions, and updating the state given that contributors to this project will likely not be primarily front end or React fluent engineers.

This provides a better description of the use of ES6 generators in the redux-saga flow: https://github.com/redux-saga/redux-saga